src/aliceVision/system/main.hpp

```diff
@@ -0,0 +1,44 @@
+// This file is part of the AliceVision project.
+// Copyright (c) 2020 AliceVision contributors.
+// This Source Code Form is subject to the terms of the Mozilla Public License,
+// v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+/**
+ * @file \c main() function wrapper
+ * Provides an implementation of \c main() that automatically catches and logs
+ * otherwise unhandled exceptions.
+ *
+ * To use this wrapper you need to change your source file containing \c main() as such:
+ * 1. Include this header
+ * 2. Rename \c main() to \c aliceVision_main()
+ */
+
+#include "Logger.hpp"
+
+#include <stdexcept>
+
+/**
+ * @brief Name of the application entry function, replacing \c main().
+ */
+int aliceVision_main(int argc, char* argv[]);
+
+// Inline to ensure the right main() is used (libf2c also has one...)
```

Comment:
I would suggest a more explicit comment, like:
```
// Implementation of the unique main entry point.
// This method will call aliceVision_main and catch the exceptions to
// log the error message and avoid seg fault on exception on windows.
```

Reply:
Ok, I expanded the comment.